### PR TITLE
Diamond pattern

### PR DESCRIPTION
**Pull request by kkumuda**
### Describe your change:
added code to create a diamond pattern based on the given input

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/diamond_pattern.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+rows = int(input("Enter the number of rows: "))  
+  
+# It is used to print the space  
+k = 2 * rows - 2  
+# Outer loop to print number of rows  
+for i in range(0, rows):  
+    # Inner loop is used to print number of space  
+    for j in range(0, k):  
+        print(end=" ")  
+    # Decrement in k after each iteration
```
Comment (IdoErel) [inline other/diamond_pattern.py:10]: Unecessary documentation/comment.

**Review by IdoErel — CHANGES_REQUESTED** [review]
Many unnecessary variable assignments.

Encapsulate code with functions, and define variables and return types.

Usage of lists is strongly suggested (instead of assigning many variables of the same type).

- - -

Don't rely on a specific format of the input.
Remove trailing white spaces, and handle input & conversion errors.


Kind regards,
IdoErel

**Comment by tianyizheng02** [comment]
What's the purpose of the txt files? One of them contains just contains the same source code as `pythagorean_theorem.py` (albeit with a bug) and the other one is empty

**Comment by tianyizheng02** [comment]
Speaking of which, 3 of these commits are for an entirely different algorithm, so they should be separated into another PR @kkumuda

**Comment by L3str4nge** [comment]
I am closing this PR because there are actually few separate algorithms with weird `.txt` files. Please reopen further.

**Pull request closed without merging**